src/main/java/edu/harvard/iq/dataverse/engine/command/impl/CreateDatasetCommand.java

```diff
@@ -136,15 +136,16 @@ public Dataset execute(CommandContext ctxt) throws CommandException {
                 && theDataset.getGlobalIdCreateTime() == null) {
             if (protocol.equals("doi")) {
                 String doiRetString = "";
+                IdServiceBean idServiceBean = IdServiceBean.getBean(ctxt);
                 try{
                     logger.log(Level.FINE,"creating identifier");
-                    doiRetString = IdServiceBean.getBean(ctxt).createIdentifier(theDataset);
+                    doiRetString = idServiceBean.createIdentifier(theDataset);
                 } catch (Exception e){
                     logger.log(Level.WARNING, "Exception while creating Identifier: " + e.getMessage(), e);
                 }
 
                 // Check return value to make sure registration succeeded
-                if (doiProvider.equals("EZID") && doiRetString.contains(theDataset.getIdentifier())) {
+                if (!idServiceBean.registerWhenPublished() && doiRetString.contains(theDataset.getIdentifier())) {
```

Comment:
- [ ] For the AbstractIdServiceBean, please add a ```postDeleteCleanup``` cleanup method that will be called by the ```DestroyDatasetCommand```.  
  - ```public void postDeleteCleanup(){ ...}```
  - [ ]  Within the ```AbstractIdServiceBean```, ```postDeleteCleanup``` can be added as a method that does nothing.  
  - [ ] For the DataCite implementation, override the ```postDeleteCleanup``` method and add the DataCite cleanup